packages/ics721/src/execute.rs

```diff
@@ -258,8 +259,24 @@ where
         let instantiate = if CLASS_ID_TO_NFT_CONTRACT.has(deps.storage, class.id.clone()) {
             vec![]
         } else {
+            let class_id = ClassId::new(class.id.clone());
```

Comment:
part 1 of instantiate2: using class_id as salt,

Reply:
Smart!